packages/ui-chart-components/src/types.ts

```diff
@@ -54,7 +55,12 @@ export type PresentationOptions = {
   hideAverage?: boolean;
   valueFormat?: string;
   conditions?: ConditionalMatrixConditionShape[];
-  referenceLines?: any;
+  referenceLines?: {
+    targetLine?: {
+      referenceLabel: string;
+      referenceValue: number;
+    };
+  };
```

Comment:
Getting rid of the `any` here seems to be causing a problem in [PieChart.tsx:145](https://github.com/beyondessential/tupaia/blob/a8715bb3cd78584c88328d7df1c321a44aff6933/packages/ui-chart-components/src/components/Charts/PieChart.tsx#L145). It should be a pretty simple fix, but can whoever review help explain this error in a more intern-friendly way? 😅

![Screenshot 2024-02-02T164202](https://github.com/beyondessential/tupaia/assets/33956381/7721f72e-c234-4e1e-a883-88cb6d5d19be)

Reply:
`any` is a bit of a wildcard in typescript, which is why it's best avoided. Any type that unions with `any` will itself become `any`. In that area of code we are referencing an unknown value of a key in the `PresentationOptions`. Before, since that value could possibly be `any` the type gets treated as `any`. But now that we've supplied a specific type, the `any` doesn't cascade through and the underlying type un-safety is exposed.

I took a stab at re-writing that function to be typesafe (at the cost of being more verbose):
```
  const getPresentationOption = (key: keyof PresentationOptions | string, option: string) => {
    if (!presentationOptions) {
      return undefined;
    }

    if (!(key in presentationOptions)) {
      return undefined;
    }

    const keyValue = presentationOptions[key as keyof PresentationOptions];
    if (typeof keyValue !== 'object') {
      return undefined;
    }

    if (!(option in keyValue)) {
      return undefined;
    }

    return keyValue[option as keyof typeof keyValue];
  };
```

Happy to pair on this if you want to understand the `any` effect better, it's a bit surprising

Reply:
I’ll give this a go, and let you know if I need to take you up on pairing—thanks!

My gut feels that the presentation options type should be defined properly from the get-go, and remain intact as it comes downstream, but there are perhaps more moving parts than I was able to grok when I was working here

Reply:
Turns out the trouble come from where we _use_ `getPresentationOption()` was being _called_

TypeScript reckons in [PieChart.tsx:156](https://github.com/beyondessential/tupaia/blob/a8715bb3cd78584c88328d7df1c321a44aff6933/packages/ui-chart-components/src/components/Charts/PieChart.tsx#L156) it returns `undefined`, which makes assignment in the next line 👎

```ts
let label = getPresentationOption(...); // type undefined
if (!label) label = name;               // `name` is a string
```

Fixed this, but should I go ahead with your type-safer version of the function?